### PR TITLE
Proposed fixes for a couple of MAVLink-related bugs

### LINKED ISSUE #33: Setting vehicle position twice requires QGC
If QGC is not running, then pressing the SET VEHICLE POSITION button twice will fail. This is because `DvlDriver.set_current_position()` is trying to get `/VFR_HUD/message/alt`, but the autopilot is not sending VFR_HUD messages.

I tested this on a Pi3 + Pixhawk bench setup with no external sensors. I recently re-installed BlueOS.

To repro:
1. Install the WL DVL extension
2. Select SET VEHICLE POSITION twice. The first time `DvlDriver.has_origin_set()` will return false and a SET_GPS_GLOBAL_ORIGIN message will be sent. The 2nd time `DvlDriver.has_origin_set()` will return true, but no GLOBAL_VISION_POSITION_ESTIMATE message will be sent.

From the docker container logs:
~~~
2023-09-04 20:28:59.239 | WARNING  | dvl:has_origin_set:218 - Unable to read current time for GPS_GLOBAL_ORIGIN, using 0: could not convert string to float: 'None'
2023-09-04 20:28:59.241 | DEBUG    | dvl:has_origin_set:222 - Trying to read origin, try # 0
2023-09-04 20:28:59.800 | INFO     | dvl:set_current_position:245 - Origin has already been set, sending POSITION_ESTIMATE instead
[2023-09-04 20:28:59,808] ERROR in app: Exception on /setcurrentposition/47.60764953320023/-122.34408152105063 [GET]
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 2311, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 1834, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 1737, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.9/site-packages/flask/_compat.py", line 36, in reraise
    raise value
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 1832, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 1818, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/pi/dvl-a50/main.py", line 96, in set_current_position
    return str(api.set_current_position(lat, lon))
  File "/home/pi/dvl-a50/main.py", line 56, in set_current_position
    return self.dvl.set_current_position(float(lat), float(lon))
  File "/home/pi/dvl-a50/dvl.py", line 248, in set_current_position
    depth = float(self.mav.get("/VFR_HUD/message/alt"))
ValueError: could not convert string to float: 'None'
192.168.2.1 - - [04/Sep/2023 20:28:59] "GET /setcurrentposition/47.60764953320023/-122.34408152105063 HTTP/1.1" 500 -
~~~

I think the fix is to add something like this to `DvlDriver.setup_mavlink()`:
~~~
        self.mav.ensure_message_frequency("VFR_HUD", 74, 5)
~~~

### LINKED ISSUE #36: VISION_POSITION_DELTA.angle_delta should be radians
MAVLink ArduPilotMega `VISION_POSITION_DELTA.angle_delta` field should be radians, but `mavllink2resthelper.send_vision` sends degrees.

https://github.com/bluerobotics/BlueOS-Water-Linked-DVL/blob/7bb1bb5214922bdd546289ef9405f5a99ebb8113/dvl-a50/mavlink2resthelper.py#L319

`mavllink2resthelper.send_vision_position_estimate` does the conversion correctly.

I found this when investigating if / how we can fuse the DVL compass readings with the Navigator compass readings.

### PR DESCRIPTION
**Pull request by ES-Alexander**
Fixes #33 
Fixes #36

Untested (I don't have the device)

### CODE REVIEW

**Comment by ES-Alexander** [comment]
@Williangalvani / @clydemcqueen it would be useful if you could review and/or test this :-)

I don't seem to have permission to request reviewers.

**Comment by clydemcqueen** [comment]
Looks good to my eye, but getting access to the device to test it might take a while. (It's winter up here. :-)

**Comment by clydemcqueen** [comment]
As it turns out, the Seattle Aquarium flew one of their ROVs for some testing today, so I got a chance to try out this PR.

TL;DR this PR fixes both bugs, but I found a new one.

For #33, I was able to set the vehicle position w/o QGC running. Nice!

For #36 I _think_ that the VPD messages contain radian deltas vs degree deltas, but the subtraction logic isn't normalized correctly, so the deltas can be outside of the range [-pi, pi].

See: https://github.com/bluerobotics/BlueOS-Water-Linked-DVL/blob/master/dvl-a50/dvl.py#L370

The result of this calculation should be normalized so that the delta is in the range [-180, 180]. Then when you convert to radians later it will be in the range [-pi, pi].

One small request: it might be nice to use variable names or comments to keep track of what units we're using. :-)

This is all in support of fusing DVL compass in the ArduSub EKF, but looking at the data it isn't obvious to me that this will be useful. When I do a naive summation of the deltas the results don't make sense. I might be missing something here. I think the right approach here is to do this on the bench w/ a small tub of water, something that I won't be able to do for a while.

All that being said, I think that this PR is an improvement over the current functionality, so LGTM.

**Comment by ES-Alexander** [comment]
Just saw this - thanks for the review @clydemcqueen :-)

> I _think_ that the VPD messages contain radian deltas vs degree deltas

That is indeed [what the MAVLink spec says](https://mavlink.io/en/messages/ardupilotmega#VISION_POSITION_DELTA), at least.

> The result of this calculation should be normalized so that the delta is in the range [-180, 180]. Then when you convert to radians later it will be in the range [-pi, pi].

Hmm, I'm unsure on this - in my mind normalisations should generally occur in the accumulator (in this case in the autopilot's receiving code) rather than on the deltas, since if the accumulator state needs to stay within a certain range then it should already be normalising (e.g. 180 + 1 will still cause a wrap around, so even a minimal delta could require an accumulator normalisation), and conceptually a delta could in fact be multiple revolutions.

@Williangalvani do you have any thoughts on this? I can add normalisation if you think it's relevant, but otherwise feel free to merge the PR (assuming you don't see anything else wrong with it) :-)

> One small request: it might be nice to use variable names or comments to keep track of what units we're using. :-)

That does seem helpful for avoiding issues like this in the first place, although that's then a pretty fiddly change to the codebase, which is somewhat confused by the mavlink2resthelper doing unit conversions internally, which means the main code is using units that don't match the MAVLink spec.

I think I'd prefer to leave that for a later PR, but I can try to add it if necessary, it'll just require some careful checking / testing afterwards.

**Comment by clydemcqueen** [comment]
Good point @ES-Alexander -- the accumulator will have to normalize anyway, and so the 2pi and -2pi spikes should not affect the result any.

Merge away! Thanks for making these fixes!

**Comment by clydemcqueen** [comment]
The Seattle Aquarium is still running pr37 from my dockerhub, so I will post install instructions for them and possibly others:

_Extension Identifier_
~~~
clydemcqueen.blueos-water-linked-dvl
~~~

_Extension Name_
~~~
blueos-water-linked-dvl
~~~

_Docker image_
~~~
clydemcqueen/blueos-water-linked-dvl
~~~

_Docker tag_
~~~
pr37
~~~

_Custom settings_
~~~
{ "NetworkMode": "host" }
~~~

**Comment by Williangalvani** [comment]
> The Seattle Aquarium is still running pr37 from my dockerhub, so I will post install instructions for them and possibly others:
> 


The only thing holding this back, from my side, was the lack of testing. I guess we can get it in now =]
